crates/telio-nurse/src/heartbeat.rs

```diff
@@ -1199,4 +1208,19 @@ mod tests {
         assert_eq!(true, meshnet_enabled);
         assert_eq!("vpn:7600617f9f9db5691a8c2768bd9d8110:2", external_links);
     }
+
+    #[tokio::test]
+    async fn test_send_analytics_report_once_on_skipped_ticks() {
+        let State { mut analytics, .. } = setup();
+
+        // First tick should not fail. After this task_interval should catch up with the skipped ticks.
+        if let Err(_) = timeout(Duration::from_millis(200), analytics.task_interval.tick()).await {
+            assert_eq!(true, false);
+        }
+
+        // Ticks should have caught up and this should timeout
+        if let Ok(_) = timeout(Duration::from_millis(200), analytics.task_interval.tick()).await {
+            assert_eq!(true, false);
+        }
+    }
```

Comment:
This doesn't check the change in bevaiour of sending messages, only that some property changed. Please write a test that verifies the change in the bevaiour.

If you are unsure if it is possible to observe this in test, here is a **hacked** together test that does that: https://github.com/NordSecurity/libtelio/commit/8e51a57dbc8c183142a87457c62179145ad72b39

Reply:
Done